packaging/Makefile_packaging.mk

```diff
@@ -0,0 +1,445 @@
+	rm -f ./v*.tar.{gz,bz*,xz}
+	$(SPECTOOL) -g $(SPEC)
+
+$(DL_VERSION).tar.$(SRC_EXT): $(SPEC) $(CALLING_MAKEFILE)
+	rm -f ./*.tar.{gz,bz*,xz}
+	$(SPECTOOL) -g $(SPEC)
+
+$(DEB_TOP)/%: % | $(DEB_TOP)/
+
+$(DEB_BUILD)/%: % | $(DEB_BUILD)/
+
+$(DEB_BUILD).tar.$(SRC_EXT): $(notdir $(SOURCE)) | $(DEB_TOP)/
+	ln -f $< $@
+
+$(DEB_TARBASE).orig.tar.$(SRC_EXT): $(DEB_BUILD).tar.$(SRC_EXT)
+	rm -f $(DEB_TOP)/*.orig.tar.*
+	ln -f $< $@
+
+deb_detar: $(notdir $(SOURCE)) $(DEB_TARBASE).orig.tar.$(SRC_EXT)
+	# Unpack tarball
+	rm -rf ./$(DEB_TOP)/.patched ./$(DEB_TOP)/.detar
+	rm -rf ./$(DEB_BUILD)/* ./$(DEB_BUILD)/.pc ./$(DEB_BUILD)/.libs
+	mkdir -p $(DEB_BUILD)
+	tar -C $(DEB_BUILD) --strip-components=1 -xpf $<
+
+# Extract patches for Debian
+$(DEB_TOP)/.patched: $(PATCHES) check-env deb_detar | \
+	$(DEB_BUILD)/debian/
+	mkdir -p ${DEB_BUILD}/debian/patches
+	mkdir -p $(DEB_TOP)/patches
+	for f in $(PATCHES); do \
+          rm -f $(DEB_TOP)/patches/*; \
+	  if git mailsplit -o$(DEB_TOP)/patches < "$$f" ;then \
+	      fn=$$(basename "$$f"); \
+	      for f1 in $(DEB_TOP)/patches/*;do \
+	        [ -e "$$f1" ] || continue; \
+	        f1n=$$(basename "$$f1"); \
+	        echo "$${fn}_$${f1n}" >> $(DEB_BUILD)/debian/patches/series ; \
+	        mv "$$f1" $(DEB_BUILD)/debian/patches/$${fn}_$${f1n}; \
+	      done; \
+	  else \
+	    fb=$$(basename "$$f"); \
+	    cp "$$f" $(DEB_BUILD)/debian/patches/ ; \
+	    echo "$$fb" >> $(DEB_BUILD)/debian/patches/series ; \
+	    if ! grep -q "^Description:\|^Subject:" "$$f" ;then \
+	      sed -i '1 iSubject: Auto added patch' \
+	        "$(DEB_BUILD)/debian/patches/$$fb" ;fi ; \
+	    if ! grep -q "^Origin:\|^Author:\|^From:" "$$f" ;then \
+	      sed -i '1 iOrigin: other' \
+	        "$(DEB_BUILD)/debian/patches/$$fb" ;fi ; \
+	  fi ; \
+	done
+	touch $@
+
+
+# Move the debian files into the Debian directory.
+ifeq ($(ID_LIKE),debian)
+$(DEB_TOP)/.deb_files: $(shell find $(TOPDIR)/debian -type f) deb_detar | \
+	  $(DEB_BUILD)/debian/
+	cd $(TOPDIR)/ && \
+	    find debian -maxdepth 1 -type f -exec cp '{}' '$(BUILD_PREFIX)/$(DEB_BUILD)/{}' ';'
+	if [ -e $(TOPDIR)/debian/source ]; then \
+	  cp -r $(TOPDIR)/debian/source $(DEB_BUILD)/debian; fi
+	if [ -e $(TOPDIR)/debian/local ]; then \
+	  cp -r $(TOPDIR)/debian/local $(DEB_BUILD)/debian; fi
+	if [ -e $(TOPDIR)/debian/examples ]; then \
+	  cp -r $(TOPDIR)/debian/examples $(DEB_BUILD)/debian; fi
+	if [ -e $(TOPDIR)/debian/upstream ]; then \
+	  cp -r $(TOPDIR)/debian/upstream $(DEB_BUILD)/debian; fi
+	if [ -e $(TOPDIR)/debian/tests ]; then \
+	  cp -r $(TOPDIR)/debian/tests $(DEB_BUILD)/debian; fi
+	rm -f $(DEB_BUILD)/debian/*.ex $(DEB_BUILD)/debian/*.EX
+	rm -f $(DEB_BUILD)/debian/*.orig
+ifneq ($(GIT_INFO),)
+	cd $(DEB_BUILD); dch --distribution unstable \
+	  --newversion $(DEB_PREV_RELEASE)$(GIT_INFO) \
+	  "Git commit information"
+endif
+	touch $@
+endif
+
+# see https://stackoverflow.com/questions/2973445/ for why we subst
```

Comment:
You might want to explain at a high level because the website could change any day.

Reply:
The code is inherit from packaging, cant really tell you what is this about.